src/NuGet.Clients/NuGet.PackageManagement.UI/PackageLoadContext.cs

```diff
@@ -58,30 +57,30 @@ public async Task<IEnumerable<string>> GetSupportedFrameworksAsync()
 
             foreach (var project in Projects)
             {
-                (bool targetFrameworkSuccess, NuGetFramework framework) = await project.TryGetMetadataAsync<NuGetFramework>(NuGetProjectMetadataKeys.TargetFramework, CancellationToken.None);
-                if (targetFrameworkSuccess)
+                IProjectMetadataContextInfo projectMetadata = await project.GetMetadataAsync(CancellationToken.None);
+                NuGetFramework framework = projectMetadata.TargetFramework;
+
+                if (framework != null)
                 {
-                    if (framework != null
-                        && framework.IsAny)
+                    if (framework.IsAny)
                     {
                         // One of the project's target framework is AnyFramework. In this case,
                         // we don't need to pass the framework filter to the server.
                         return Enumerable.Empty<string>();
                     }
 
-                    if (framework != null
-                        && framework.IsSpecificFramework)
+                    if (framework.IsSpecificFramework)
                     {
                         frameworks.Add(framework.DotNetFrameworkName);
```

Comment:
Out of scope for this PR, but use of `NuGetFramework.DotNetFameworkName` can have unexpected results with .NET 5 TFMs when platforms are used, like `net5.0-windows`.

If you see other usage of it, it may be worthwhile to take note. @zkat, perhaps we need to do a find all references to `DotnetFrameworkName` and determine which ones might cause NuGet to not work correctly in .NET 5.